### PR TITLE
Bump Go to 1.7.1

### LINKED ISSUE #2516: Update Go version to 1.7+
Docker has already moved on to 1.7 and we're still at 1.5.4, which is causing issues for some new PRs, including test failures.

cc @dongluochen @vieux @allencloud

### PR DESCRIPTION
**Pull request by dongluochen**
Close #2516.

Signed-off-by: Dong Chen <dongluo.chen@docker.com>

### CODE REVIEW

**Comment by dongluochen** [comment]
Break it into 2 parts. First part is to update CI, https://github.com/docker/swarm/pull/2522. This one will update after CI update.

**Comment by dongluochen** [comment]
@nishanttotla @allencloud PTAL. This should unblock other PRs. Next step is to clean up network test failure with engine-master, which can be done in #2492.

**Comment by nishanttotla** [comment]
LGTM

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -23,9 +23,9 @@ If you don't already have `git` installed, you should install it.  For example,
 sudo apt-get install git
 ```
 
-You also need Go 1.5 or higher.  Download Go from [https://golang.org/dl/](https://golang.org/dl/).  To install on Linux:
+You also need Go 1.6 or higher.  Download Go from [https://golang.org/dl/](https://golang.org/dl/).  To install on Linux:
```
Comment (dongluochen) [inline CONTRIBUTING.md]: 1.6 should work. But we haven't done testing. Let's use 1.7.

**Pull request merged**